xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-templates/src/main/resources/saverights.vm

```diff
@@ -28,8 +28,15 @@
 
 #if ($!{services.csrf.isTokenValid("$!{request.getParameter('form_token')}")})
 #foreach($obj in $doc.getObjects($clsname)) ## XWiki.XWikiGlobalRights or XWiki.XWikiRights
+  ## In subwikis, users or groups may be referenced by "wiki:Space.Username" or "Space.Username"
+  #set ($wikiname = $xcontext.database)
+  #if ($wikiname == "xwiki")
```

Comment:
You shouldn't hard-code "xwiki". If you want to treat the main wiki differently then we have ``$xcontext.mainWikiName``.

Reply:
Well noted as well. The latest commit proposes a more generic approach which does not treat the main wiki differently, so I'm resolving this conversation.